### ISSUE TITLE
Ordering of Msg constructors

### ISSUE THREAD

**Issue by KasMA1990** [avh4/elm-format#823]
In my day-to-day writing Elm code, I can only think of one significant place where I don't have autoformatting today, but wish I had: specifying the order of my `Msg` constructors in my `update` functions. That is, if my `update` function looks like this:

```elm
update msg model =
    case msg of
        A -> ...
        B -> ...
        C -> ...
```

then I know that the `A,B,C` ordering is something I've defined somewhere.

This usually comes up when I want to add new constructors to an existing `Msg` type. Often those new constructors relate to some existing messages, so I probably want to group them. The most natural place to do so, I find, is with the type definition itself. So although a type like:

```elm
type Msg 
    = A
    | B
    | C
```

looks fine, and it's easy to keep `update` using the same ordering, it becomes more annoying to keep them in sync when the type has morphed into

```elm
type Msg 
    = A
    -- Initialisation messages 
    | B
    | D
    -- Drag and drop impl messages
    | C
    | E
    | F
...
```

and it would be really nice if there was a way to just keep `update` in sync with an ordering like this. Some caveats though:
- I think I would only want this for `Msg` types and the `update` function, and not in general for all custom types everywhere.
- I get that this may not be the right decision for everyone, so it might be worth having as something you opt into.

Of course, those two points can make it tricky to unite this feature with `elm-format`, because it eschews configuration and special cases.

On the other hand, `elm-format` already has this feature for a different kind of ordering: formatting exports as specified in module docs. So maybe there is room for something similar for `Msg` and `update`? There's lots of room for bikeshedding here, so I won't make concrete proposals for how it should look before testing if this is something anyone besides me would be interested in 😊

**Comment by avh4** [comment]
My initial thought is that this is probably a bit beyond the scope of elm-format, as I'm a bit more hesitant to move blocks of code around than I am about the existing features of sorting imports and `exposing` clauses.  However, I'm open to further discussion.

Also, FYI, there appears to be an existing [elm-review](https://package.elm-lang.org/packages/jfmengels/elm-review/latest/) rule that implements this https://package.elm-lang.org/packages/SiriusStarr/elm-review-no-unsorted/1.1.4/NoUnsortedCases/ so you could try that out and see if it either already solves your problem, or use it to learn a bit more about the details of how this should best behave.

**Comment by jfmengels** [comment]
I agree that this is probably not best suited for `elm-format`, if only for the fact that saving might introduce large changes in an editor when running `elm-format`.

Limiting this to functions named `update` (what about derivatives like `updateHelp`?) is also non-`elm-format` like. I agree that @SiriusStarr's `elm-review` rule is probably a better solution (though I have not tried it out before).

**Comment by SiriusStarr** [comment]
Yep, this is exactly what `NoUnsortedCases` does, so that should work fine.  Please let me know if you have questions or run into any issues with it.
